histomicstk/features/ComputeHaralickFeatures.py

```diff
@@ -0,0 +1,234 @@
+    computer vision," Journal of Open Research Software, vol 1, 2013.
+    """
+
+    # List of feature names
+    feature_list = [
+        'Haralick.ASM.Mean',
+        'Haralick.Contrast.Mean',
+        'Haralick.Correlation.Mean',
+        'Haralick.SumofSquar.Mean',
+        'Haralick.IDM.Mean',
+        'Haralick.SumAverage.Mean',
+        'Haralick.SumVariance.Mean',
+        'Haralick.SumEntropy.Mean',
+        'Haralick.Entropy.Mean',
+        'Haralick.Variance.Mean',
+        'Haralick.DifferenceEntropy.Mean',
+        'Haralick.IMC1.Mean',
+        'Haralick.IMC2.Mean',
+        'Haralick.ASM.Range',
+        'Haralick.Contrast.Range',
+        'Haralick.Correlation.Range',
+        'Haralick.SumofSquar.Range',
+        'Haralick.IDM.Range',
+        'Haralick.SumAverage.Range',
+        'Haralick.SumVariance.Range',
+        'Haralick.SumEntropy.Range',
+        'Haralick.Entropy.Range',
+        'Haralick.Variance.Range',
+        'Haralick.DifferenceEntropy.Range',
+        'Haralick.IMC1.Range',
+        'Haralick.IMC2.Range',
+    ]
+
+    # compute object properties if not provided
+    if rprops is None:
+        rprops = regionprops(im_label)
+
+    # create pandas data frame containing the features for each object
+    numFeatures = len(feature_list)
+    numLabels = len(rprops)
+    fdata = pd.DataFrame(np.zeros((numLabels, numFeatures)),
+                         columns=feature_list)
+
+    # check for consistent shapes between 'I' and 'Label'
+    if im_intensity.shape != im_label.shape:
+        raise ValueError("Inputs 'I' and 'Label' must have same shape")
+
+    # determine if image is grayscale or RGB
+    if len(im_intensity.shape) != 2:  # color image
+        raise ValueError("Inputs 'I' should be a grayscale image")
+
+    for i in range(numLabels):
+
+        # get bounds of an intensity image
+        minr, minc, maxr, maxc = rprops[i].bbox
+
+        # grab nucleus mask
+        subImage = im_intensity[minr:maxr+1, minc:maxc+1].astype(np.uint8)
+
+        # gets GLCM or gray-tone spatial dependence matrix
+        arrayGLCM = graycomatrixext(subImage, offsets=offsets,
+                                    num_levels=num_levels,
+                                    gray_limits=gray_limits,
+                                    symmetric=True, normed=True)
+        num_dims = len(subImage.shape)
+
+        # offsets
+        if offsets is None:
+            # set default offset value
+            offsets = np.identity(num_dims)
+        else:
+            # check sanity
+            if offsets.shape[1] != num_dims:
+                raise ValueError(
+                    'Dimension mismatch between input image and offsets'
+                )
+        num_offsets = offsets.shape[0]
+        f = np.zeros((numFeatures/2, num_offsets))
+        for r in range(num_offsets):
+
+            # normalize GLCM
+            R = np.sum(arrayGLCM[:, :, r], dtype=np.float)
+            nGLCM = arrayGLCM[:, :, r]/R
```

Comment:
You should not have to do this anymore. Calling `graycomatrixext` with `normed=True` should already give you normalized glcms.